src/main/scala/ClockUtil.scala

```diff
@@ -165,21 +165,21 @@ class ClockDivider(width: Int) extends Module {
 
 // Performs clock division when divisor >= 1, as done in ClockDivider
 // When divisor is 0, pass through the clock
-class ClockDivideOrPass(width: Int, depth: Int = 3, genClockGate: () => ClockGate) extends Module {
+class ClockDivideOrPass(width: Int, depth: Int = 3, genClockGate: () => ClockGate) extends RawModule {
   val io = IO(new Bundle {
-    val divisor = Input(UInt(width.W))
+    val clockIn = Input(Clock())
+    val divisor = Input(UInt(width.W)) // this may not be synchronous with clockIn
     val resetAsync = Input(AsyncReset())
     val clockOut = Output(Clock())
 
   })
-
-  val divider = Module(new ClockDivider(width))
-  divider.io.divisor := io.divisor
+  val divider = withClockAndReset(io.clockIn, io.resetAsync) { Module(new ClockDivider(width, initDiv=1)) }
+  divider.io.divisor := withClockAndReset(io.clockIn, io.resetAsync) { SynchronizerShiftReg(Mux(io.divisor === 0.U, 1.U, io.divisor)) }
```

Comment:
The Mux seems unnecessary here. We can just gate the divider since the clock mux below is passing the clock thru when divisor === 0.U.

Reply:
Unfortunately not, the `ClockMutexMux` needs the clocks to be alive, else it can get stuck waiting on a dead clock that never advances its synchronizer-shift-reg..